tensorpipe/test/CMakeLists.txt

```diff
@@ -50,42 +56,50 @@ if(TP_ENABLE_IBV)
 endif()
 
 if(TP_ENABLE_CMA)
-  target_sources(tensorpipe_test PRIVATE
+  list(APPEND TP_TEST_PRIVATE_SRCS
     channel/cma/cma_test.cc
     )
   add_subdirectory(channel/cma)
 endif()
 
+add_executable(tensorpipe_test ${TP_TEST_PUBLIC_SRCS})
+target_sources(tensorpipe_test PRIVATE ${TP_TEST_PRIVATE_SRCS})
```

Comment:
Why define the target before we've even finished determining its sources? This seems to defeat the purpose of the `TP_TEST_..._SRCS` variables, as in, if we're going to incrementally add sources to the target then we could just define the target at the beginning and just drop those variables.

Reply:
We can do this way too, but I feel we have a cleaner way now with the list of files and added to `add_executable() `and `target_sources()` removed.